mate-panel/panel-toplevel.c

```diff
@@ -2000,40 +1997,40 @@ get_delta (int       src,
 static void
 panel_toplevel_update_animating_position (PanelToplevel *toplevel)
 {
-	GTimeVal   time_val;
+	GDateTime *now;
 	int        deltax, deltay, deltaw = 0, deltah = 0;
 	int        monitor_offset_x, monitor_offset_y;
 
-	g_get_current_time (&time_val);
+	now = g_date_time_new_now_local ();
```

Comment:
Why do you need `GDateTime` here?

Reply:
The current datetime is used for calculating the ratio in a time interval, which can also be seen as the slope. It determines the acceleration of the hide/show transition.

Reply:
I know what code does... I mean why do you need to use `GDateTime` for this? Why not use suggested replacement that will do exactly same as function you are replacing?

https://gitlab.gnome.org/GNOME/glib/-/blob/master/glib/gmain.c#L2768
https://gitlab.gnome.org/GNOME/glib/-/blob/master/glib/gmain.c#L2799

Reply:
Now I understand what you mean, you are asking me, why I use an object (GDateTime) instead of a basic type (gint64). I think the methods (g_date_time_difference, g_date_time_add) of the object (GDateTime) are useful and do not represent an unusual consumption of resources.

I also saw there is no need to do any time unit conversion, as the time unity is lost in division:
`x = (double)n/d;` being n and d time_differences.

Reply:
```python
import numpy as np
import matplotlib.pyplot as plt

t_start = 0
t_end = 100

def ratio(t):
	n = t - t_start;
	d = t_end - t_start; # it does not depend on the time varible: t
	return n/d;

def slope(r):
	return -2 * (r*r) * (r-1.5);

t = np.arange(t_start, t_end, 1)

plt.plot(t, ratio(t), 'r--', t, slope(ratio(t)), 'bs', t, slope(slope(ratio(t))), 'g^')
plt.ylabel('Visibility')
plt.xlabel('Time')
plt.show()
```
![Figure_1](https://user-images.githubusercontent.com/10171411/79963784-1c196100-848a-11ea-83aa-082cf56f7119.png)